### PR TITLE
feat(react): create useOrderShippingAddressChangeRequest hook

### PR DESCRIPTION
**Pull request by andreecosta**
## Description

This adds the useOrderShippingAddressChangeRequest hook.

### Dependencies

None.

## Checklist

- [x] The commit message follows our guidelines
- [x] Tests for the respective changes have been added
- [x] The code is commented, particularly in hard-to-understand areas
- [x] The labels and/or milestones were added

## Disclaimer

By sending us your contributions, you are agreeing that your contribution is made subject to the terms of our [Contributor Ownership Statement](https://github.com/Farfetch/.github/blob/master/COS.md)

### CODE REVIEW

File: packages/react/src/orders/hooks/useOrderShippingAddressChangeRequests.ts [hunk 1]
```diff
@@ -0,0 +1,264 @@
...
+    }
+    case actionTypes.ResetOrderShippingAddressChangeRequestsState: {
+      return initialState;
+    }
+    default:
+      return state;
+  }
+}
+
+function useOrderShippingAddressChangeRequests(
```
Comment (boliveira) [inline packages/react/src/orders/hooks/useOrderShippingAddressChangeRequests.ts:143]: Does it make sense to add this hook to the `useOrder` hook and expose its actions from there as well?

**Review by pedro-gbf — APPROVED** [review]

File: packages/react/src/orders/hooks/__tests__/useOrderShippingAddressChangeRequests.test.ts [hunk 2]
```diff
@@ -0,0 +1,362 @@
...
+const mockFetchConfig = {
+  myCustomParameter: 10,
+};
+
+describe('useOrderShippingAddressChangeRequests', () => {
+  beforeEach(jest.clearAllMocks);
+
+  afterEach(cleanup);
+
+  it('should return correctly with initial state and call all hook dependencies with the correct options', () => {
```
[the remark this answers is not in the document]
Comment (andreecosta) [reply packages/react/src/orders/hooks/__tests__/useOrderShippingAddressChangeRequests.test.ts]: Done, changed the message.

File: packages/react/src/orders/hooks/useOrderShippingAddressChangeRequests.ts (hunk 1 above)
Comment (andreecosta) [reply packages/react/src/orders/hooks/useOrderShippingAddressChangeRequests.ts:143]: You are right, it does make sense. Exposed the actions on `useOrder` hook.

Comment (boliveira) [reply packages/react/src/orders/hooks/useOrderShippingAddressChangeRequests.ts:143]: Can you change the useOrder test to check if the `useOrderShippingAddressChangeRequest` hook is called with the correct parameters, please?

Comment (andreecosta) [reply packages/react/src/orders/hooks/useOrderShippingAddressChangeRequests.ts:143]: Sorry, missed it. Fixed!

**Review by boliveira — APPROVED** [review]

**Pull request merged**